src/EventListener/PostQueryBuilderSubscriber.php

```diff
@@ -170,6 +179,16 @@ protected function filterQueryBuilder(QueryBuilder $queryBuilder, string $field,
                 $parameter = null;
                 $filterDqlPart = $field.' IS NOT NULL';
                 break;
+            // Special case if value is a ListFilter
+            case $value instanceof ListFilter:
+                if (null === $value->getValue()  || '' === $value->getValue() ) {
+                    // Break if there is not value
+                    break;
+                }
+
+                $filterDqlPart = $field.' ' .$value->getOperator() .' :'.$parameter;
```

Comment:
IMO, QueryBuilder is responsible of selecting the ORM operator matching data returned by `ListFilter::getOperator` method